src/main/java/org/opensearch/sdk/ExtensionsRunner.java

```diff
@@ -56,52 +57,79 @@
 import static org.opensearch.common.UUIDs.randomBase64UUID;
 
 public class ExtensionsRunner {
+    private ExtensionSettings extensionSettings = getExtensionSettings();
+    private DiscoveryNode opensearchNode;
```

Comment:
Created setter/getter to access opensearchNode.

Reply:
Getter and setter are ok (but unnecessary with all-private access) but the original question stands.  It is initialized here to `null` and not set otherwise until the call to `handlePluginsRequest()`.

In `handleIndicesModuleRequest()` you retrieve this value, and in theory it could be null, where it would then be passed to `transportService.connectToNode()` (and later,`transportService.sendRequest()`).  That will probably throw an NPE [here](https://github.com/opensearch-project/OpenSearch/blob/7c73ed8d4c65eee6433e9895036bca1eae4c16e2/server/src/main/java/org/opensearch/transport/TransportService.java#L431).

So should there be a check whether that is null/uninitialized, and a more helpful error message to the user?

Also are there tests that show the preferred order in which these steps are taken?

Reply:
That makes sense. So `handlePluginsRequest` will the first request to the SDK from OpenSearch for the transport communication. This request will basically initialize the extension and will be a part of OpenSearch bootstrap. Rest of the request will require a trigger through APIs on OpenSearch side. Tests will covered as a part of https://github.com/opensearch-project/opensearch-sdk/issues/25

Reply:
Got it.  I'm going to approve this PR but I marked the spot where I would want a log warning. Or some comment indicating it's needed. Javadocs. Something.  Consider that an "approved with suggestion" as I know you'll be doing more work on this later.